core/src/main/scala/org/apache/spark/network/buffer/LargeByteBufferInputStream.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.network.buffer
+
+import java.io.InputStream
+
+/**
+ * Reads data from a LargeByteBuffer, and optionally cleans it up using buffer.dispose()
+ * when the stream is closed (e.g. to close a memory-mapped file).
+ */
+private[spark]
+class LargeByteBufferInputStream(private var buffer: LargeByteBuffer, dispose: Boolean = false)
```

Comment:
sure, I can change it, but just curious -- why?  worried that it some scalaisms will sneak in that hurt performance?  Or for consistence w/ the other LargeByteBuffer classes?  (I figure this will mostly get *used* from scala code, so I though consistency was a wash either way.)